src/contract.ts

```diff
@@ -18,26 +27,26 @@ export class Contract {
             Object.defineProperty(this, methodName, {
                 writable: false,
                 enumerable: true,
-                value: async function(args: any) {
-                    if (arguments.length > 1) {
+                value: nameFunction(methodName, async (args: object = {}, ...ignored) => {
+                    if (ignored.length || Object.prototype.toString.call(args) !== '[object Object]') {
                         throw new PositionalArgsError();
                     }
-                    return this.account.viewFunction(this.contractId, methodName, args || {});
-                }
+                    return this.account.viewFunction(this.contractId, methodName, args);
+                })
             });
         });
         changeMethods.forEach((methodName) => {
             Object.defineProperty(this, methodName, {
                 writable: false,
                 enumerable: true,
-                value: async function(args: any, gas?: BN, amount?: BN) {
-                    if (arguments.length > 3) {
+                value: nameFunction(methodName, async (args: object = {}, gas?: BN, amount?: BN) => {
+                    if (Object.prototype.toString.call(args) !== '[object Object]') {
```

Comment:
it's still useful to check for argument count as well in case first argument of contract call is also object (though neither guarantees absence of false negatives)

Reply:
I added a test for this, and I moved related tests from `test/account.test.js` to `test/contract.test.js`